internal/engine/compiler/impl_amd64.go

```diff
@@ -4420,14 +4421,19 @@ func (c *amd64Compiler) compileCallFunctionImpl(index wasm.Index, codeAddressReg
 	// So we increment the call frame stack pointer.
 	c.assembler.CompileNoneToMemory(amd64.INCQ, amd64ReservedRegisterForCallEngine, callEngineGlobalContextCallFrameStackPointerOffset)
 
+	if amd64CallingConventionModuleInstanceAddressRegister == targetFunctionAddressRegister {
```

Comment:
This is the key, others are just styles

Reply:
Just wondering, is the reason to only mov in this branch an optimization to save a register for other use?

Reply:
yeah, optimization as not always end up in this situation, and the less instructions the better in general!